### PR TITLE
Configure securityContext, resources, and probes in the deployments

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: deployments/engine/templates/deployment.yaml [hunk 1]
```diff
@@ -69,6 +73,12 @@ spec:
         - name: config
           mountPath: /etc/config
           readOnly: true
+        - name: tmp
```
Comment (kkaneda) [inline deployments/engine/templates/deployment.yaml:76]: Could you add some comment to explain why this change is needed? My understanding is that this is to load the models in memory. If so, I think this should be fine, but we just need to be a bit careful to make sure if we have sufficient memory when multiple models are loaded (this only apply to Ollama, though).

**Review by kkaneda — APPROVED** [review]
LG!

File: deployments/engine/templates/deployment.yaml (hunk 1 above)
Comment (kkaneda) [reply deployments/engine/templates/deployment.yaml:76]: Ah nvm. I misunderstood the behavior of `emptyDir`.

File: deployments/engine/templates/deployment.yaml [hunk 2]
```diff
@@ -69,6 +73,12 @@ spec:
...
+          mountPath: /tmp
+        - name: tmp
+          mountPath: /.ollama
```
Comment (guangrui-cloudnatix) [inline deployments/engine/templates/deployment.yaml:79]: Can we rename this to something like `/.llm` or `/.engine`, since this should be applicable to the engine with `vLLM as well?

**Review by guangrui-cloudnatix — APPROVED** [review]

File: deployments/engine/templates/deployment.yaml (hunk 2 above)
Comment (Ladicle) [reply deployments/engine/templates/deployment.yaml:79]: This name depends on ollama behavior, so if another engine writes something to the root filesystem, we need to add the path here.
ollama generates the key at startup in `/.ollama`, and this change allows ollama to write to it.

Comment (guangrui-cloudnatix) [reply deployments/engine/templates/deployment.yaml:79]: I see.  Thanks for the explanation. I'm not sure if vllm needs to write root filesystem.
